python/dgl/distributed/rpc_client.py

```diff
@@ -161,17 +162,17 @@ def connect_to_server(ip_config, num_servers, max_queue_size=MAX_QUEUE_SIZE, net
     for server_id, addr in server_namebook.items():
         server_ip = addr[1]
         server_port = addr[2]
-        rpc.add_receiver_addr(server_ip, server_port, server_id)
-    rpc.sender_connect()
+        while not rpc.connect_receiver(server_ip, server_port, server_id):
+            time.sleep(1)
     # Get local usable IP address and port
     ip_addr = get_local_usable_addr(server_ip)
     client_ip, client_port = ip_addr.split(':')
+    # wait server connect back
+    rpc.receiver_wait(client_ip, client_port, num_servers, blocking=False)
```

Comment:
Will it cause problem if it's not blocking here? Thus the message might be sent out below before the remote sender connect to local receiver?

Reply:
it's ok. because sender is independent of receiver for the same client. it's ok to send message before remote sender connects to local receiver. The remote sender sends messages only after connects to local receiver. And local receiver will be blocked in rpc.recv_response() until remote sender connects to local receiver and send messges.